src/client.ts

```diff
@@ -19,103 +25,179 @@ export class APIClient {
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
+    const endpoint = `${BASE_URI}/corps`;
+
+    await this.getJson(endpoint, {
+      headers: this.generateHeaders(),
+    }).catch((error) => {
+      if (error.status === 401) {
+        throw new IntegrationProviderAuthenticationError({
+          cause: error,
+          endpoint,
+          status: error.status,
+          statusText: error.statusText,
+        });
+      } else {
+        throw new IntegrationProviderAPIError({
+          cause: error,
+          endpoint,
+          status: error.status,
+          statusText: error.statusText,
+        });
+      }
     });
+  }
 
-    try {
-      await request;
-    } catch (err) {
-      throw new IntegrationProviderAuthenticationError({
-        cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
-        status: err.status,
-        statusText: err.statusText,
-      });
+  /**
+   * Iterates each corp resource in the provider.
+   *
+   * @param iteratee receives each resource to produce entities/relationship
+   */
+  public async iterateCorps(
+    iteratee: ResourceIteratee<SignalSciencesCorp>,
+  ): Promise<void> {
+    const endpoint = `${BASE_URI}/corps`;
+
+    const { data } = await this.getJsonWithRetries(endpoint, {
+      headers: this.generateHeaders(),
+    });
+
+    for (const corp of data) {
+      await iteratee(corp);
     }
   }
 
   /**
-   * Iterates each user resource in the provider.
+   * Iterates each user resource in the provider based on the provided corp.
    *
    * @param iteratee receives each resource to produce entities/relationships
    */
   public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
+    corpName: string,
+    iteratee: ResourceIteratee<SignalSciencesUser>,
   ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
+    const endpoint = this.buildEndpoint(corpName, '/users');
+
+    const { data } = await this.getJsonWithRetries(endpoint, {
```

Comment:
make sure to add a TODO here for supporting pagination. You will almost certainly need it for iterating through users.

Reply:
Pagination is available within the API but is not supported on the users/corps endpoints at this time.

Reply:
🤯  well then... Maybe someday they'll have more than one page of users 😉 .
Can we document this somewhere? Perhaps in your `types` for user and corps respectively.